base/applications/mspaint/registry.cpp

```diff
@@ -189,39 +191,32 @@ void RegistrySettings::Store()
 
 void RegistrySettings::SetMostRecentFile(LPCTSTR szPathName)
 {
+    /* Register the file to the system 'Recent' folder */
     if (szPathName && szPathName[0])
         SHAddToRecentDocs(SHARD_PATHW, szPathName);
 
-    if (strFile1 == szPathName)
+    /* If szPathName was found in strFiles, then bring it to the top of strFiles */
+    for (INT i = MAX_RECENT_FILES - 1, iFound = -1; i > 0; --i)
     {
-        // do nothing
+        if (iFound < 0 && strFiles[i].CompareNoCase(szPathName) == 0)
+            iFound = i;
+
+        if (iFound >= 0)
+        {
+            CString tmp = strFiles[i];
+            strFiles[i] = strFiles[i - 1];
+            strFiles[i - 1] = tmp;
+        }
     }
-    else if (strFile2 == szPathName)
-    {
-        CString strTemp = strFile2;
-        strFile2 = strFile1;
-        strFile1 = strTemp;
-    }
-    else if (strFile3 == szPathName)
-    {
-        CString strTemp = strFile3;
-        strFile3 = strFile2;
-        strFile2 = strFile1;
-        strFile1 = strTemp;
-    }
-    else if (strFile4 == szPathName)
-    {
-        CString strTemp = strFile4;
-        strFile4 = strFile3;
-        strFile3 = strFile2;
-        strFile2 = strFile1;
-        strFile1 = strTemp;
-    }
-    else
+
+    /* If szPathName is not the top of strFiles, then insert szPathName at top of strFiles  */
+    if (strFiles[0].CompareNoCase(szPathName) != 0)
     {
-        strFile4 = strFile3;
-        strFile3 = strFile2;
-        strFile2 = strFile1;
-        strFile1 = szPathName;
+        for (INT i = MAX_RECENT_FILES - 1; i > 0; --i)
+        {
+            strFiles[i] = strFiles[i - 1];
+        }
```

Comment:
curly braces could be omitted here, allowing for more code to fit the screen vertically at once.